tools/shared.py

```diff
@@ -136,92 +137,39 @@ def returncode_to_str(code):
   return f'returned {code}'
 
 
-# Runs multiple subprocess commands.
-# bool 'check': If True (default), raises an exception if any of the subprocesses failed with a nonzero exit code.
-# string 'route_stdout_to_temp_files_suffix': if not None, all stdouts are instead written to files, and an array of filenames is returned.
-# bool 'pipe_stdout': If True, an array of stdouts is returned, for each subprocess.
 def run_multiple_processes(commands,
                            env=None,
                            route_stdout_to_temp_files_suffix=None,
                            pipe_stdout=False,
                            check=True,
                            cwd=None):
+  """Runs multiple subprocess commands.
+
+  check : bool
+    If True (default), raises an exception if any of the subprocesses
+    failed with a nonzero exit code.
+
+  route_stdout_to_temp_files_suffix : string
+    if not None, all stdouts are instead written to files, and an array
+    of filenames is returned.
+
+  pipe_stdout : bool
+    If True, an array of stdouts is returned, for each subprocess.
+  """
+  # On windows, there is limit on the total number of tasks which is
+  # enforced bu multiprocessing, but only on newer version of python:
+  # https://bugs.python.org/issue26903
+  # TODO(sbc): This could be removed once we update the min python version to 3.8.0
+  global multiprocessing_pool
+  if not multiprocessing_pool:
+    max_workers = get_num_cores()
+    if WINDOWS:
+      max_workers = min(max_workers, 61)
```

Comment:
Is the fix just this, to limit to 61? So this was only ever a problem for machines with >61 cores..?

Reply:
Yes, that is my understanding.  see the python bug for details.  I guess for emcc users it would also effect users who set EMCC_CORES > 61.. since you can create more threads that cores if you want to using that variable.b

Reply:
Hmm, I'm a little surprised then that we got but reports about this. How many people have that many cores..?

If we can confirm this works, or we can ask someone that has encountered the bug, that would make me more confident here. But, maybe I'm being overly cautious, if this is in upstream Python it's probably fine..?

Reply:
Yes, but it makes sense it you look at the original bug report : "On the other Windows system, this issue does not seem to ever occur. The affected system is a 64-thread Threadripper 2990WX, and the numbers are quite close to 64, so I wonder if that has a meaning here. The unaffected system is a 16-thread system."